.github/workflows/build.yml

```diff
@@ -65,34 +65,82 @@ jobs:
       run: echo 'cmake --build ${{github.workspace}}/build --target bootcd --target livecd' | ${{github.workspace}}/RosBE-CI/RosBE.sh . 0 ${{matrix.arch}}
     - name: Print ccache statistics
       run: ccache -s
-    - name: Upload bootcd
+    - name: Upload ISOs
       uses: actions/upload-artifact@v2
       with:
-        name: reactos-bootcd-${{matrix.compiler}}-${{matrix.arch}}-${{github.sha}}
-        path: build/bootcd.iso
-    - name: Upload livecd
+        name: reactos-${{matrix.compiler}}-${{matrix.arch}}-${{github.sha}}
+        path: |
+          build/bootcd.iso
+          build/livecd.iso
+  
+  build-msvc:
+    strategy:
+      matrix:
+        arch: [i386, amd64]
+        toolset: ['14.2', '14.1'] # VS 2019, 2017
+        include:
+        - arch: i386 # Not compiling on amd64 prompt
+          toolset: '14.0' # VS 2015
+      fail-fast: false
+    runs-on: windows-latest
+    steps:
+    - name: Install ninja
+      run: choco install -y ninja
+    - name: Install Flex & Bison
+      run: |
+        curl -O https://svn.reactos.org/storage/vperevertkin/flexbison.7z
+        7z x flexbison.7z -O${{github.workspace}}\bin
+        echo "${{github.workspace}}\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+        echo "BISON_PKGDATADIR=${{github.workspace}}\bin\share\bison" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+        echo "M4=${{github.workspace}}\bin\m4.exe" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+    - name: Activate VS cmd (x86)
+      if: ${{ matrix.arch == 'i386' }}
+      uses: ilammy/msvc-dev-cmd@v1
+      with:
+        arch: amd64_x86
+        toolset: ${{ matrix.toolset }}
+    - name: Activate VS cmd (amd64)
+      if: ${{ matrix.arch == 'amd64' }}
+      uses: ilammy/msvc-dev-cmd@v1
+      with:
+        arch: amd64
+        toolset: ${{ matrix.toolset }}
+    - name: Source checkout
+      uses: actions/checkout@v2
+      with:
+        path: src
+    - name: Configure
+      run: cmake -S src -B build -G Ninja -DCMAKE_TOOLCHAIN_FILE:FILEPATH=toolchain-msvc.cmake -DARCH:STRING=${{matrix.arch}} -DENABLE_ROSTESTS=1 -DENABLE_ROSAPPS=1
+    - name: Build
+      run: cmake --build ${{github.workspace}}\build
+    - name: Generate ISOs
+      run:  cmake --build ${{github.workspace}}\build --target bootcd --target livecd
+    - name: Upload ISOs & symbols
       uses: actions/upload-artifact@v2
       with:
-        name: reactos-livecd-${{matrix.compiler}}-${{matrix.arch}}-${{github.sha}}
-        path: build/livecd.iso
+        name: reactos-msvc${{matrix.toolset}}-${{matrix.arch}}-${{github.sha}}
+        path: | 
+          build/bootcd.iso
+          build/livecd.iso
+          build/msvc_pdb
 
-  build-msvc:
+
+  build-clang-cl:
```

Comment:
Maybe you could still do with if/include/exclude? (I'm not sure which one is preferable.)

Reply:
I don't really know. I am also happy to not cripple this with a lot of ifs